flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/logical/FlinkLogicalTableSourceScan.scala

```diff
@@ -104,6 +105,21 @@ class FlinkLogicalTableSourceScan(
       s"Scan($s)"
     }
   }
+
+  override def estimateRowCount(mq: RelMetadataQuery): Double = {
+    val tableSourceTable = getTable.unwrap(classOf[TableSourceTable[_]])
+
+    if (tableSourceTable.getStatistic != FlinkStatistic.UNKNOWN) {
```

Comment:
Same as above. We somehow need to incorporate the selectivity of a pushed-down filter.